src/space/objects/realization/blackhole.js

```diff
@@ -0,0 +1,92 @@
+import Type from "../pattern/encounterObjectType";
+import MyMath from "../../utils/myMath";
+import AssetLoader from "../../utils/assetLoader";
+
+const BH_SPRITE_RADIUS = 50;
+const BH_RENDER_RADIUS = 50;
+const BH_FORCE_SCALE = 6.0/10.0;
+const BH_GRAVITY_RANGE = 8000;
+const BH_WRAP_INTERVAL = 100;
+
+export default class BlackHole extends EncounterObject{
+
+    /**
+      * Construction for black hole game element
+      *
+      * @param {Encounter} encounter
+      * @param {Vector} position - black hole's position
+      * @reference Black Hole (Warp) by Timothy Courtney (timothycourtney.io)
+      *  Licensed under Creative Commons: By Attribution 3.0 License
+      *  http://creativecommons.org/licenses/by/3.0/
+      */
+    constructor(encounter, position){
+        super(encounter, Type.BLACK_HOLE, BH_GRAVITY_RANGE, position);
+        this.bhImage = AssetLoader.getAsset("black-hole");
+        this.coordinateXInImage = 0;
+        this.coordinateYInImage = 0;
+        this.wrapInterval = BH_WRAP_INTERVAL;
+    }
+
+    /**
+      * generate a random position blackhole
+      *
+      * @param {Object} encounter
+      * @return {BlackHole} a new black hole with random position
+      */
+    static generateAtRandomPosition(encounter){
+        let x = Math.floor(Math.random()*encounter.width);
+        let y = Math.floor(Math.random()*encounter.height);
+        return new BlackHole(encounter, new Vector(x,y));
+    }
+
+    /**
+      * @param {Object} the object waiting for check
+      * @return {Boolean} whether other object collided with black hole
+      */
+    collidesWith(others){
+        let distance = MyMath.distance(this.position, others.position);
+        return !(distance > BH_GRAVITY_RANGE);
+    }
+
+    /**
+      * Draw the black hole image in the map
+      *
+      * @param {CanvasRenderingContext2D} context - context to render content on
+      * @param {DOMHighResTimeStamp} elapsedTime - time elapsed from last frame
+      */
+    render(deltaT, context){
+        this.wrapInterval -= deltaT;
+        if(this.wrapInterval <= 0){
+            if(this.coordinateXInImage == 500 && this.coordinateYInImage == 100){
+                this.coordinateXInImage = 0;
+                this.coordinateYInImage = 0;
+            }
+            else if(this.coordinateXInImage == 500){
+                this.coordinateYInImage += 100;
+    	        this.coordinateXInImage = 0;
+            }
+            else{
+                this.coordinateXInImage += 100;
+            }
+            this.wrapInterval = BH_WRAP_INTERVAL;
+        }
+        //change the display size according to the radiusScale
+        let renderRadius = BH_RENDER_RADIUS * this.radiusScale;
+        context.drawImage(this.bhImage,this.coordinateXInImage,this.coordinateYInImage,2*BH_SPRITE_RADIUS,2*BH_SPRITE_RADIUS,this.position.x - renderRadius, this.position.y - renderRadius, 2*renderRadius, 2*renderRadius);
```

Comment:
This then can be also refactored to:

```
context.drawImage( this.bhImage, this.coordinateXInImage, this.coordinateYInImage,
BS_SPRITE_WIDTH, BH_SPRITE_HEIGHT, this.position.x - renderRadius, this.position.y - renderRadius, 
2*renderRadius, 2*renderRadius );
```

Also please consider adding some line breaks to this extremely long line ;)

Reply:
Ok, this takes away some flexibility, but it is certainly ok :)